src/identity/identity.service.ts

```diff
@@ -31,6 +32,84 @@ export class IdentityService {
     this.logger.setContext(IdentityService.name);
   }
 
+  /**
+   * @async
+   * Retrieve an identity by userId and providerType.
+   * @param {string} userId - the userId of the wanted identity
+   * @param {ProviderType} providerType - the providerType of the wanted identity
+   */
+  async getIdentityFromUserIdAndProviderType(
+    userId: string,
+    providerType: ProviderType,
+  ): Promise<Identity> {
+    const identity = await this.identityRepository.findOne({
+      where: {
+        providerUserId: userId,
+        providerType: providerType,
+      },
+      relations: ['user'],
+    });
+    if (identity === undefined) {
+      throw new NotInDBError(`Identity for user id '${userId}' not found`);
+    }
+    return identity;
+  }
+
+  /**
+   * @async
+   * Sync the given Identity with the given information
+   * @param {Identity} identity - the identity to update
+   * @param {string | undefined} displayName - the displayName to update the user with
+   * @param {string | undefined} email - the email to update the user with
+   * @param {string | undefined} profilePicture - the profilePicture to update the user with
+   */
+  async syncIdentity(
```

Comment:
I think `syncIdentity` is a bit confusing because its only one way and to over network or something
```suggestion
  async updateIdentity(
```

Reply:
changed